tools/shared.py

```diff
@@ -217,6 +217,13 @@ def check_call(cmd, *args, **kw):
   CONFIG_FILE = None
   logging.debug('EM_CONFIG is specified inline without a file')
 else:
+  gen_config = False
+  try:
```

Comment:
Thanks for the PR!

I think we should look for `--generate-config` as an argument to emcc. See where we check for `--help`, `--version` etc. in `emcc.py` (early in `run()`). (`shared.py` is loaded in other places than emcc, and that argument is not relevant in those places)

Might need to move the code that emits the config file into a function, that is called from either this file or from `emcc.py`.